modules/standard/IO.chpl

```diff
@@ -3899,6 +3953,107 @@ proc channel.readline(ref arg: ?t): bool throws where t==string || t==bytes {
   return true;
 }
 
+/*
+  Read a line into a Chapel string. Reads until a ``\n`` is reached.
+
+  :arg arg: a string to receive the line
+  :arg maxSize: The maximum amount of characters to read.
+  :arg stripNewline: Whether to strip the trailing ``\n`` from the line.
+  :returns: `true` if a line was read without error, `false` upon EOF
+
+  :throws SystemError: Thrown if data could not be read from the channel.
+*/
+proc channel.readLine(ref s: string, maxSize=-1, stripNewline=false): bool throws{
+  if writing then compilerError("read on write-only channel");
+  const origLocale = this.getLocaleOfIoRequest();
+
+  try {
+    on this.home {
+      try this.lock(); defer { this.unlock(); }
+      var saveStyle: iostyleInternal = this._styleInternal();
+      defer {
+        this._set_styleInternal(saveStyle);
+      }
+      var myStyle = saveStyle.text();
+      myStyle.string_format = QIO_STRING_FORMAT_TOEND;
+      myStyle.string_end = 0x0a; // ascii newline.
+      if(maxSize >= 0){
+        try myStyle.max_width_characters = maxSize.safeCast(uint(32)); // Probably should change maxSize's type to be safe to cast
+      }
+      this._set_styleInternal(myStyle);
+      try _readOne(iokind.dynamic, s, origLocale);
+    }
+  } catch err: SystemError {
+    if err.err != EEOF then throw err;
+    return false;
+  }
+  // TODO find a more memory efficient way to do this
+  // Maybe have a QIO_STRING_FORMAT_TOEND_DROPEND which drops the last char
+  if(stripNewline){
+    s = s.strip(chars="\n", leading = false);
+  }
+
+  return true;
+}
+
+/*
+  Read a line into Chapel bytes. Reads until a ``\n`` is reached.
+
+  :arg arg: bytes to receive the line
+  :arg maxSize: The maximum amount of bytes to read.
```

Comment:
how about "The maximum number of bytes to read."

Reply:
Also needs to say that "the default of -1 means no limit on the number of bytes to read".